src/Middleware/ParametersNameDecodingHandler.cs

```diff
@@ -0,0 +1,53 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+
+/// <summary>
+/// This handlers decodes special characters in the request query parameters that had to be encoded due to RFC 6570 restrictions names before executing the request.
+/// </summary>
+public class ParametersNameDecodingHandler: DelegatingHandler
+{
+    /// <summary>
+    /// The options to use when decoding parameters names in URLs
+    /// </summary>
+    public ParametersNameDecodingOption EncodingOptions { get; set; }
```

Comment:
maybe private or internal?
It is ideally set through the constructor and there's the potential path through the request options of the message itself.

Reply:
```suggestion
    internal ParametersNameDecodingOption EncodingOptions { get; set; }
```